### PR TITLE
MM-8853: Adding MANAGE_EMOJIS and MANAGE_OTHERS_EMOJIS permissions

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add the Manage Emojis permissions, including tests and data migration.

#### Ticket Link
[MM-8853](https://mattermost.atlassian.net/browse/MM-8853)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has enterprise changes (https://github.com/mattermost/enterprise/pull/301)
- [x] Touches critical sections of the codebase (auth, upgrade, etc.): It includes a small data migration.

### CODE REVIEW

File: api4/emoji.go [hunk 1]
```diff
@@ -54,6 +48,28 @@ func createEmoji(c *Context, w http.ResponseWriter, r *http.Request) {
...
 
+	// Allow any user with MANAGE_EMOJIS permission at Team level to manage emojis at system level
+	memberships, err := c.App.GetTeamMembersForUser(c.Session.UserId)
+
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_EMOJIS) {
```
Comment (grundleborg) [inline api4/emoji.go]: I don't think this outer check is necessary as the inner checks below will all cascade up to it.

File: app/app.go [hunk 2]
```diff
@@ -603,3 +594,62 @@ func (a *App) SetPhase2PermissionsMigrationStatus(isComplete bool) error {
...
+	system := model.System{
+		Name:  EMOJIS_PERMISSIONS_MIGRATION_KEY,
+		Value: "true",
+	}
+
+	if result := <-a.Srv.Store.System().Save(&system); result.Err != nil {
+		mlog.Critical("Failed to mark emojis permissions migration as completed.")
+		mlog.Critical(fmt.Sprint(result.Err))
+	}
+}
```
Comment (grundleborg) [inline app/app.go]: We're not setting `MANAGE_OTHERS_EMOJIS` anywhere in this migration. Is that intentional?

Comment (jespino) [reply app/app.go]: Yes, was intentional, but I'm thinking is needed as system_admin user, by default.

Comment (grundleborg) [reply app/app.go]: :+1: to giving it to system admin by default.

File: api4/emoji.go (hunk 1 above)
Comment (jespino) [reply api4/emoji.go]: Fixed

**Comment by grundleborg** [comment]
@jespino `TestDeleteEmoji` is failing, but otherwise looks great.

File: api4/emoji.go (hunk 1 above)
Comment (jespino) [reply api4/emoji.go]: I realized that there is a corner case we weren't considering. If you are system admin but you aren't a team member.

**Pull request merged**